php/proxy.php

```diff
@@ -3,150 +3,192 @@
-     
+
+        if($_SERVER['REQUEST_METHOD'] == 'DELETE'){
+            $postData = self::getPOSTdata();
+            curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'DELETE');
+            curl_setopt($ch, CURLOPT_POSTFIELDS,$postData);
+        }
+
+        $headers = self::getHeadersOfRequest();
+
         curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
         curl_setopt($ch, CURLOPT_HEADER, 1);
         curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
-								
-        $response = curl_exec ($ch);
+
+        //Uncomment below 2 lines to disable SSL verification
+
+        //curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, 0);
+        //curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, 0);
+
+
+        //Uncomment and set certificate path for SSL verification.
+
+        //curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, true);
+        //curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, 2);
+        //curl_setopt($ch, CURLOPT_CAINFO, getcwd() . "path to .crt file");
+
+        $response = curl_exec($ch);
+
         if (curl_error($ch)){
             print curl_error($ch);
         } else {
             self::processResponse($response);
         }
         curl_close ($ch);
     }
-    
+
     private function processResponse($response) {
         $response = str_replace("HTTP/1.1 100 Continue\r\n\r\n","",$response);
         $responseArray = explode("\r\n\r\n", $response, 2);
         $headers = $responseArray[0];
         $result = $responseArray[1];
         $headerArray = explode(chr(10), $headers);
-        
+
         foreach($headerArray as $header=>$headerValue){
             if(!preg_match("/^Transfer-Encoding/", $headerValue)){
-                $headerValue = str_replace($targetPath, $domainURL, $headerValue);
+                $headerValue = str_replace($this->targetPath, $this->domainURL, $headerValue);
                 header(trim($headerValue));
             }
         }
-        
-        if(strpos($requestURI,'report.xml') !== false){
-            echo $result;
-        } else {
-            $result = str_replace($targetPath, $domainURL, $result);
-            print $result;
-        }
-    }
-    
-    public function initialize() {
-        self::disable_gzip();
-        $settings = self::getSettings();
-        $targetPath = $settings['url'];
-        if(isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] == 'on'){
-            $domainURL = 'https://'.$_SERVER['HTTP_HOST'];
-        } else {
-            $domainURL = 'http://'.$_SERVER['HTTP_HOST'];
-        }
-        $part = '/acrolinx/proxy.php'; //TODO: Ensure that this is the correct location of the 'proxy.php' on the server.
-        $requestURI = $_SERVER['REQUEST_URI'];
-        $pos = self::getPosition($requestURI, $part);
-        if ($pos !== false) {
-            $requiredURI = self::getActualURI($requestURI, $pos, $part);
-            $url = $targetPath.$requiredURI;
-        }
-        self::proxyRequest($url); 
+
+        $result = str_replace($this->targetPath, $this->domainURL, $result);
```

Comment:
I think we should not patch the result...
It looks quite dangerous to me...
Shouldn't that be already correct because of the X Acrolinx Base Url header introduced with server 5.2?

Reply:
Fix: Removing the patch. I had look at the documentation at https://acrolinxapi.docs.apiary.io/#